### ISSUE TITLE
A false negative about the rule StringInstantiation

### ISSUE THREAD

**Issue by Belle-PL** [pmd/pmd#3848]
**Affects PMD Version:6.42.0** 

**Rule:StringInstantiation**

Please provide the rule name and a link to the rule documentation:
<https://pmd.github.io/latest/pmd_rules_java_performance.html#stringinstantiation>

**Description:**

Hi, I found a false negative about the rule **StringInstantiation**. Please read the following code example.  PMD should have reported a warning at line 5 because this line create a String object by `new`. Hence, I think this is a false negative. Thanks for your kind consideration.

**Code Sample demonstrating the issue:**

```java
String getLiteral() {
    return "foo";
}
String foo() {
    return new String(getLiteral()); // should report a warning at this line
}
```

**Expected outcome:Warning**

PMD should report a violation at line 5, but doesn't. This is a false-negative.

**Running PMD through:** *[Maven]*

**Comment by jsotuyod** [comment]
This is fixed in PMD 7.0.0-rc1. It won't be backported to PMD 6.
